File: backend/FarmaciaEliseu.Api/Models/Produto.cs
namespace FarmaciaEliseu.Api.Models;

public class Produto
{
    public int ProdutosId { get; set; }
    public string? Descricao { get; set; }
    public string? UnidadeMedida { get; set; }
    public decimal? PrecoCompra { get; set; }
    public decimal? PrecoVenda { get; set; }
    public string? Localizacao { get; set; }
    public string? Laboratorio { get; set; }
    public string? Principio { get; set; }
    public string? Generico { get; set; }
    public string? CodigoProduto { get; set; }
    public string? CodigoBarras { get; set; }
    public DateTime DataCadastro { get; set; }
    public DateTime DataUltimoRegistro { get; set; }
    public bool Deletado { get; set; }
}

File: frontend/src/types/Produto.ts
export interface Produto {
  produtosId: number
  descricao: string | null
  unidadeMedida: string | null
  precoCompra: number | null
  precoVenda: number | null
  localizacao: string | null
  laboratorio: string | null
  principio: string | null
  generico: string | null
  codigoProduto: string | null
  codigoBarras: string | null
  dataCadastro: string
  dataUltimoRegistro: string
}

export interface ProdutoInput {
  descricao: string
  unidadeMedida?: string | null
  precoCompra?: number | null
  precoVenda?: number | null
  localizacao?: string | null
  laboratorio?: string | null
  principio?: string | null
  generico?: string | null
  codigoProduto?: string | null
  codigoBarras?: string | null
}
